boa/src/object/internal_methods.rs

```diff
@@ -902,15 +902,23 @@ impl GcObject {
         // 4. For each element key of ownKeys, do
         for key in own_keys {
             // a. If Type(key) is String, then
-            if let PropertyKey::String(key_str) = &key {
+            let key_str = if let PropertyKey::Index(index) = &key {
+                Some(index.to_string().into())
+            } else if let PropertyKey::String(key_str) = &key {
+                Some(key_str.clone())
+            } else {
+                None
+            };
+
+            if let Some(key_str) = key_str {
                 // i. Let desc be ? O.[[GetOwnProperty]](key).
                 let desc = self.__get_own_property__(&key);
                 // ii. If desc is not undefined and desc.[[Enumerable]] is true, then
                 if let Some(desc) = desc {
                     if desc.expect_enumerable() {
                         // 1. If kind is key, append key to properties.
                         if let PropertyNameKind::Key = kind {
```

Comment:
We could also match on `PropertyNameKind` too

Reply:
Changed that in bb25549